full_bubble_model.py

```diff
@@ -510,93 +784,224 @@ def plot(cpar, t_int=np.array([0.0, 1.0]), n=5.0, base_name='', LSODA_timeout=30
+            verticalalignment='center',
+            bbox={'facecolor': 'white', 'pad': 0, 'linewidth': 0.0},
+        )
+
+    # plot settings
+    plt.ylim([1e-24, 5.0*max_mol[indexes_to_plot[-1]]])
+    ax.set_yscale('log')
+    if num_sol.t[end_index] < 1e-3:
+        ax.set_xlabel('$t$ [μs]')
+    else:
+        ax.set_xlabel('$t$ [ms]')
     ax.set_ylabel('$n_k$ [mol]')
-    ax.grid()
-    ax.legend()
+    if not presentation_mode: ax.grid()
+    if show_legend: ax.legend()
 
     plt.show()
+
+# plot pressure excitation
+    if plot_pressure:
+        plt.rcParams.update({'font.size': 24 if presentation_mode else 18})
+        fig3 = plt.figure(figsize=(16, 9) if presentation_mode else (20, 6))
+        ax = fig3.add_subplot(axisbelow=True)
+        ex_args = np.array([cpar[name] for name in excitation_args], dtype=np.float64)
+        external_pressure = [1e-3*Excitation(time, cpar.P_amb, ex_args)[0] for time in num_sol.t[:end_index]] # [MPa]
+        ax.plot(t, external_pressure, color='orange', label='external pressure', linewidth = 2.0 if presentation_mode else 1.0)
+
+        if num_sol.t[end_index] < 1e-3:
+            ax.set_xlabel('$t$ [μs]')
+        else:
+            ax.set_xlabel('$t$ [ms]')
+        ax.set_ylabel('Pressure excitation [kPa]')
+        if not presentation_mode: ax.grid()
+
+        plt.show()
+
+# plot pressure
+    if plot_pressure:
+        plt.rcParams.update({'font.size': 24 if presentation_mode else 18})
+        fig4 = plt.figure(figsize=(16, 9) if presentation_mode else (20, 6))
+        ax = fig4.add_subplot(axisbelow=True)
+        ax.plot(t, internal_pressure, color = 'g', label='internal pressure', linewidth = 2.0 if presentation_mode else 1.0)
+
+        if num_sol.t[end_index] < 1e-3:
+            ax.set_xlabel('$t$ [μs]')
+        else:
+            ax.set_xlabel('$t$ [ms]')
+        ax.set_ylabel('Internal pressure [MPa]')
+        ax.set_yscale('log')
+        if not presentation_mode: ax.grid()
+
+        plt.show()
     
 # saving the plots
     if base_name != '':
         try:
-            fig1.savefig(base_name + '_1.png')
-            fig2.savefig(base_name + '_2.png')
+            metadata = {key: str(data[key]) for key in data.keys()}
+            fig1.savefig(base_name + '_1.png', format='png', metadata=metadata)
+            fig2.savefig(base_name + '_2.png', format='png', metadata=metadata)
+            if plot_pressure:
+                fig3.savefig(base_name + '_3.png', format='png', metadata=metadata)
+                fig4.savefig(base_name + '_4.png', format='png', metadata=metadata)
         except:
-            print(f'Error in saving {base_name}_1.png')
+            print(print(colored(f'Error in saving {base_name}_1.png','red')))
 
 # print data
-    print_data(data)
+    print_data(cpar, data)
     return None
-
+           
 
 """________________________________Save to CSV________________________________"""
 
+def get_settings_and_info():
```

Comment:
Before a parameter study, a txt is saved with all info to reproduce the study, and also additional info. Example:
```General information:
    Created: 2023.10.24 15:30:48 (YYYY.MM.DD HH:MM:SS)
    Computer name: AronPC
    User name: aron
    Number of cores: 8
    Number of logical threads: 16
    CPU frequency:  3201.00 MHz
    RAM size:  28524.06 MB

parameters settings:
    model = chem_Otomo2018_without_O
    species = ['NH3' 'H2' 'H' 'NH2' 'NH' 'N' 'NNH' 'N2H4' 'N2H3' 'N2H2' 'H2NN' 'N2']
    number of species = 12
    number of reactions = 35

full_bubble_model settings:
    enable_heat_transfer = True
    enable_evaporation = False 
    enable_reactions = True
    enable_dissipated_energy = False
    target_specie = 'NH3' # Specie to calculate energy effiqiency
    excitation_type = 'no_excitation' # function to calculate pressure excitation
total_combinations = 2500
ranges = {
    "R_E": [
        1e-06,```